networking/client-ping.go

```diff
@@ -0,0 +1,60 @@
+package networking
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+	"strconv"
+	"strings"
+	"time"
+
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+)
+
+// FindBestEndpoint - requests against a given addr and port
+func FindBestEndpoint(reqAddr, currentHostPubKey, peerPubKey, serverName string, proxyPort int) error {
+	peerAddr, err := netip.ParseAddr(reqAddr) // begin validate
+	if err != nil {
+		return err
+	}
+	if _, err = wgtypes.ParseKey(peerPubKey); err != nil {
+		return err
+	}
+	if _, err = wgtypes.ParseKey(currentHostPubKey); err != nil {
+		return err
+	}
+	if len(serverName) == 0 {
+		return fmt.Errorf("no server provided")
+	} // end validate
+	c, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%d", reqAddr, proxyPort), reqTimeout)
+	if err != nil { // TODO: change verbosity for timeouts (probably frequent)
+		return err
+	}
+	defer c.Close()
+	sentTime := time.Now().UnixMilli()
+	msg := strings.Join([]string{
+		currentHostPubKey,
+		serverName,
+		strconv.Itoa(int(sentTime)),
```

Comment:
it would be much cleaner to encode and decode a message in struct format

Reply:
but also slower